File: vb/Examples/SubClass.vb
' 3. When no constructor is defined:
'    Class 'vb.SubClass' must declare a 'Sub New' because its base class 
'    'vb.SuperClass' does not have an accessible 'Sub New' that can be 
'    called with no arguments.
Public Class SubClass : Inherits SuperClass

  ' 3.5 right off the bat, the compiler makes us implement a constructor
  Sub New(parameter)
    MyBase.New(parameter)
  End Sub

End Class

File: csharp/Examples/SubClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp.Examples
{
  class SubClass : SuperClass
  {

    // 2. Add a constructor to SubClass that takes one parameter
    //    because the constructor with one parameter in the 
    //    SuperClass is not inherited.
    public SubClass(string str)
      : base(str)
    {
      // nothing additional to do, superclass handles constructing
    }

    // 4. Add a parameterless constructor
    public SubClass()
      : base()
    {
      // nothing additional to do, superclass handles constructing
    }
  }
}
